onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -2690,15 +2690,74 @@ def aten_index_copy(
     raise NotImplementedError()
 
 
+@torch_op("aten::index_put")
 def aten_index_put(
-    self: TensorType,
-    indices: Optional[Sequence[TensorType]],
-    values: TensorType,
+    self: TReal,
+    indices: Sequence[INT64],
+    values: TReal,
     accumulate: bool = False,
-) -> TensorType:
+) -> TReal:
     """index_put(Tensor self, Tensor?[] indices, Tensor values, bool accumulate=False) -> Tensor"""
 
-    raise NotImplementedError()
+    index = op.SequenceAt(indices, 0)  # assume indices only have 1 element
```

Comment:
I am bit unclear about the assumption above. Is this just for now? Is the input type expected to be changed to `INT64` ?

Reply:
Looks like indices is a list of 1d tensors, so INT64 may not do it?

Reply:
On this line, was there a reason we assume indices only has one element?

Reply:
From the test case, all the indices have only one element. If it contains multiple elements, we need to use for/while to do the logic. 
But for this aten op, the multiple elements can be combined to one element, for example, indices=[[0,1], [2,3]], can be combined to indices=[[0,1,2,3],].

Reply:
Would we do a sequenceConcat or something like that?

Reply:
I tried in Torch, actually it cannot work like this:
torch.index_put( self=tensor(5x5), indices=(tensor([0]), tensor([1]),), values=tensor(2,5))
so we don't know how to use more than one element in indices tuple.